src/classes/logic.ml

```diff
@@ -174,6 +164,28 @@ module Make
     let locfile, res = P.parse_raw_lazy ~filename contents in
     l, locfile, res
 
+  let parse_all ?language = function
+    | `File file ->
+      let l, _, (module P : S) =
+        match language with
+        | None -> of_filename file
+        | Some l -> of_language l
+      in
+      let locfile, res = P.parse_file_lazy file in
+      l, locfile, res
+    | `Raw (filename, l, s) ->
+      let language =
+        match language with Some _ -> language | None -> Some l
+      in
+      parse_raw_lazy ?language ~filename s
+    | `Stdin l ->
+      let filename = "<stdin>" in
+      let s = Dolmen_std.Misc.read_all ~size:1024 stdin in
```

Comment:
This works — it is what we are currently doing in Alt-Ergo — and is probably fine, but I was thinking doing this inside Dolmen would make it possible to create the `lexbuf` with `Lexing.from_channel stdin` directly, avoiding an extra copy (i.e. I was thinking of an equivalent of `parse_raw_lazy` that takes an `in_channel` rather than a string buffer — I can make that patch if you'd like).